src/common/Autocompletes/AccountListAutocomplete/AccountListAutocomplete.test.tsx

```diff
@@ -0,0 +1,137 @@
+import { AccountListAutocomplete } from './AccountListAutocomplete';
+
+const setFieldValue = jest.fn();
+
+const mockAccounts = [
+  { id: 'account-1', name: 'John Doe Ministry Account' },
+  { id: 'account-2', name: 'Jane Smith Support Account' },
+  { id: 'account-3', name: 'Global Missions Fund' },
+  { id: 'account-4', name: 'Local Church Partnership' },
+  { id: 'account-5', name: 'Student Ministry Account' },
+];
+
+describe('AccountListAutocomplete', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('shows the selected account', () => {
+    const { getByRole } = render(
+      <LocalizationProvider dateAdapter={AdapterLuxon}>
+        <ThemeProvider theme={theme}>
+          <AccountListAutocomplete
+            options={mockAccounts}
+            value={mockAccounts[0]}
+            onChange={(_, value) => {
+              setFieldValue(value);
+            }}
+          />
+        </ThemeProvider>
+      </LocalizationProvider>,
+    );
+    expect(getByRole('combobox')).toHaveDisplayValue(
+      'John Doe Ministry Account',
+    );
+  });
+
+  it('changes the selected account', async () => {
+    const { getByRole, findByRole } = render(
+      <LocalizationProvider dateAdapter={AdapterLuxon}>
+        <ThemeProvider theme={theme}>
+          <AccountListAutocomplete
+            options={mockAccounts}
+            value={mockAccounts[0]}
+            onChange={(_, value) => {
+              setFieldValue(value);
+            }}
+          />
+        </ThemeProvider>
+      </LocalizationProvider>,
+    );
+
+    expect(getByRole('combobox')).toHaveDisplayValue(
+      'John Doe Ministry Account',
+    );
+    userEvent.click(getByRole('combobox'));
+    userEvent.click(
+      await findByRole('option', {
+        name: 'Jane Smith Support Account',
+      }),
+    );
+    expect(setFieldValue).toHaveBeenCalledWith(mockAccounts[1]);
+  });
+
+  it('filters accounts based on input', async () => {
+    const { getByRole, findByRole, findAllByRole, queryByRole } = render(
+      <LocalizationProvider dateAdapter={AdapterLuxon}>
+        <ThemeProvider theme={theme}>
+          <AccountListAutocomplete
+            options={mockAccounts}
+            value={null}
+            onChange={setFieldValue}
+          />
+        </ThemeProvider>
+      </LocalizationProvider>,
+    );
+
+    const combobox = getByRole('combobox');
+    userEvent.type(combobox, 'Ministry');
+
+    const options = await findAllByRole('option');
+    // "John Doe Ministry Account" and "Student Ministry Account"
+    expect(options).toHaveLength(2);
+
+    expect(
+      await findByRole('option', { name: 'John Doe Ministry Account' }),
+    ).toBeInTheDocument();
+
+    expect(
+      await findByRole('option', { name: 'Student Ministry Account' }),
+    ).toBeInTheDocument();
+
+    await waitFor(() => {
+      expect(
+        queryByRole('option', { name: 'Global Missions Fund' }),
+      ).not.toBeInTheDocument();
+    });
+  });
```

Comment:
Because you use `await` for the options and ensure 2 are there, you don't need to await on the expects below.
```suggestion
        const options = await findAllByRole('option');
    // "John Doe Ministry Account" and "Student Ministry Account"
    expect(options).toHaveLength(2);

    expect(
      getByRole('option', { name: 'John Doe Ministry Account' }),
    ).toBeInTheDocument();

    expect(
      getByRole('option', { name: 'Student Ministry Account' }),
    ).toBeInTheDocument();

    expect(
      queryByRole('option', { name: 'Global Missions Fund' }),
    ).not.toBeInTheDocument();
  });
```